### ISSUE TITLE
installing a dependency which is actually a git repo

### ISSUE THREAD

**Issue by jney** [luarocks/luarocks#706]
Hi. I'd like to install this fork https://github.com/openresty/lua-cjson, which is not hosted on luarocks.org. 
How can i declare such dependency?

**Comment by Tieske** [comment]
untested:
```
luarocks install https://raw.githubusercontent.com/openresty/lua-cjson/master/lua-cjson-2.1devel-1.rockspec
```

**Comment by mkpankov** [comment]
For anyone who stumbles upon: command in comment https://github.com/luarocks/luarocks/issues/706#issuecomment-304618328 fails for unrelated reasons (rockspec has source specified to unavailable URL).

It actually works.

**Comment by jeremybusk** [comment]
Use this
luarocks install https://luarocks.org/manifests/skylothar/lua-resty-jwt-0.1.11-0.rockspec

**Comment by mecampbellsoup** [comment]
Note that if you are trying to install a luarock from a git repository URL, and you have your `.rockspec` changes in a _branch_ (like we did!) you need to specify that branch ref in the `source` attribute, e.g.:

```
# In pgmoon/pgmoon-dev-1.rockspec
...

source = {
  url = "git://github.com/leafo/pgmoon.git"
}
...

```

**Comment by esatterwhite** [comment]
Is it possible to specify a git repository as a dependency in the spec file?

**Comment by esatterwhite** [comment]
> Is it possible to specify a git repository as a dependency in the spec file?
> 
> ```lua
> dependencies = {
>   "git+ssh://github.com:organization/repo.git#ref"
> }
> ```

@Tieske is it possible to do something like this? ^ I haven't had any luck getting it to work

**Comment by Ismoh** [comment]
It's old but similiar:
Is it possible to do so without a rockspec file? Do i have to fork the repo and add a rockspec file?

**Comment by Tieske** [comment]
@esatterwhite  I don't think you can specify a repo as a dependency. In that case you proabbaly can create a little wrapper that creates a local rockstree, and then specify that one when installing (asuming you have a repo that is private, hence no rock)

@Ismoh rockspec is required. Luarocks has some commands that can write a rockspec for you, you could give that a try.

**Comment by Ismoh** [comment]
Thanks a ton for advise!

**Comment by eigenbom** [comment]
Is there a plan to add git/url to the dependencies and build_dependencies? Would be a nice addition.
